types/address.go

```diff
@@ -99,8 +99,16 @@ func AccAddressFromBech32(address string) (addr AccAddress, err error) {
 		return nil, err
 	}
 
-	if len(bz) != AddrLen {
-		return nil, errors.New("Incorrect address length")
+	validator := GetConfig().GetAddressValidator()
```

Comment:
I was confused for a second when reading this. Perhaps the nomenclature should read: `GetAddressVerifier`?

Reply:
How about `AddressFormatChecker`? Validator would probably be the right word in general if we're taking about schema/format validation, but I can see how it would be confusing in the context of Cosmos!

Reply:
What's wrong with `GetAddressVerifier`?

Reply:
> What's wrong with `GetAddressVerifier`?

Verifier makes me think of something that checks the DB for an address, but sure not a big deal

Reply:
> Also, what about calling this from all the Bech32 functions?

Do you mean `ValAddressFromBech32` and `ConsAddressFromBech32`? Are those also derived from pub keys? Not too familiar with how they work yet.